From 580cb5e6c0a0416062e30acd76a8ffb49f02555d Mon Sep 17 00:00:00 2001
From: Aditya Kumar Singh <quic_adisi@quicinc.com>
Date: Sat, 7 Jan 2023 22:56:24 +0530
Subject: [PATCH] mac80211: pass channel info in AWGN event

Currently during AWGN event, drivers only passes the
ieee80211_hw on which AWGN interference is detected and
the interference bitmap. The worker, iterates over each
channel contexts avaiable under the hardware and expects
only 1 channel contexts to be present. Then, with this
channel context it executes further steps.

However, with MLO, there can be more than 1 channel contexts
available under the hardware hence while iterating, it is
important to know on which channel AWGN interference was
detected.

Add logic to pass channel info as well in the AWGN detcted
event and use it properly to process further in the AWGN
event.

Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c |  3 ++-

--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -9091,7 +9091,8 @@ ath12k_wmi_dcs_awgn_interference_event(s
         }
         ar->awgn_chandef = *chandef;
 
-        ieee80211_awgn_detected(ah->hw, awgn_info.chan_bw_interference_bitmap);
+	ieee80211_awgn_detected(ah->hw, awgn_info.chan_bw_interference_bitmap,
+				chandef->chan);
 
         spin_lock_bh(&ar->data_lock);
         ar->awgn_intf_handling_in_prog = true;
